File: src/frontend/libs/location-api/src/lib/models/location.dto.ts
/**
 * Location model representing a rental location
 */
import type { PostalCode } from '@orange-car-rental/shared';
import type { LocationCode } from './location-code.type';
import type { LocationName } from './location-name.type';
import type { CityName } from './city-name.type';
import type { StreetAddress } from './street-address.type';
import type { FullAddress } from './full-address.type';
import type { LocationStatus } from './location-status.enum';

export type Location = {
  readonly code: LocationCode;
  readonly name: LocationName;
  readonly street: StreetAddress;
  readonly city: CityName;
  readonly postalCode: PostalCode;
  readonly fullAddress: FullAddress;
  readonly status?: LocationStatus;
};

File: src/backend/Services/Location/OrangeCarRental.Location.Application/DTOs/LocationDto.cs
namespace SmartSolutionsLab.OrangeCarRental.Location.Application.DTOs;

public sealed record LocationDto
{
    public required string Code { get; init; }
    public required string Name { get; init; }

    // Split address fields for frontend compatibility
    public required string Street { get; init; }
    public required string City { get; init; }
    public required string PostalCode { get; init; }

    // Computed full address for display
    public required string FullAddress { get; init; }

    public required string OpeningHours { get; init; }
    public required string Phone { get; init; }
    public required string Email { get; init; }
    public required string Status { get; init; }
}
